gtk/gtkshow.c

```diff
@@ -70,8 +74,69 @@ gtk_show_uri (GdkScreen    *screen,
   gdk_app_launch_context_set_screen (context, screen);
   gdk_app_launch_context_set_timestamp (context, timestamp);
 
-  ret = g_app_info_launch_default_for_uri (uri, (GAppLaunchContext*)context, error);
+  ret = g_app_info_launch_default_for_uri (uri, G_APP_LAUNCH_CONTEXT (context), error);
+  g_object_unref (context);
+
+  return ret;
+}
+
+/**
+ * gtk_show_uri_on_window:
+ * @parent: parent window
```

Comment:
Seems to be `nullable`... otherwise you could simplify the code and add a `g_return_val_if_fail()`